### PR TITLE
Allow INI keys with no values

### PR DESCRIPTION
**Pull request by aw1cks**
Hey! :)

Sometimes, an INI key is useful without a value.
This PR allows that scenario, where the value is null.

For example:

```
packages__pacman_conf:
  options:
    HoldPkg: pacman glibc
    XferCommand: /usr/bin/wget --passive-ftp -c -O %o %u
    Architecture: auto
    Color:
    CheckSpace:
    ParallelDownloads: 5
    SigLevel: Required DatabaseOptional
    LocalFileSigLevel: Optional
  core:
    Include: /etc/pacman.d/mirrorlist
  extra:
    Include: /etc/pacman.d/mirrorlist
  community:
    Include: /etc/pacman.d/mirrorlist
  multilib:
    Include: /etc/pacman.d/mirrorlist
```

becomes:

```
[community]
Include=/etc/pacman.d/mirrorlist

[core]
Include=/etc/pacman.d/mirrorlist

[extra]
Include=/etc/pacman.d/mirrorlist

[multilib]
Include=/etc/pacman.d/mirrorlist

[options]
Architecture=auto
CheckSpace
Color
HoldPkg=pacman glibc
LocalFileSigLevel=Optional
ParallelDownloads=5
SigLevel=Required DatabaseOptional
XferCommand=/usr/bin/wget --passive-ftp -c -O %o %u
```

### CODE REVIEW

**Comment by jtyr** [comment]
Not printing keys when their value is `null` intentional. It can be used remove the key from a default variable when set to `null`:

```yaml
test_section1_key1: value1
test_section1_key2: value2

test_section1:
  key1: "{{ test_section1_key1 }}"
  key2: "{{ test_section1_key2 }}"

test:
  section1: "{{ test_section1 }}" 
```

Remove  `key1`:


```yaml
test_section1_key1: null
```

How would Pacman react to the following setting?

```yaml
packages__pacman_conf:
  options:
    HoldPkg: pacman glibc
    XferCommand: /usr/bin/wget --passive-ftp -c -O %o %u
    Architecture: auto
    Color: ""
```

If that's not acceptable, we would have to implement some special sequence that would indicate that you don't want to print out `key = value` paid but only `key`. Something like this:

```yaml
packages__pacman_conf:
  options:
    HoldPkg: pacman glibc
    XferCommand: /usr/bin/wget --passive-ftp -c -O %o %u
    Architecture: auto
    Color: "!!!null"
```

**Comment by jtyr** [comment]
You can also workaround it like this:

```yaml
packages__pacman_conf:
  options:
    HoldPkg: pacman glibc
    XferCommand: /usr/bin/wget --passive-ftp -c -O %o %u
    Architecture: auto
    "# Color with no value": "\nColor"
```

**Comment by aw1cks** [comment]
Latest commit implements an escape sequence `!!!null`

File: filter_plugins/config_encoders.py [hunk 1]
```diff
@@ -408,9 +408,14 @@ def encode_ini(
                 item = '""'
 
             if item is not None:
-                rv += "%s%s%s%s%s%s\n" % (
-                    indent, prop, delimiter, quote, _escape(item, quote),
-                    quote)
+                if item == "!!!null":
+                    rv += "%s%s\n" % (indent, prop)
+                else:
```
Comment (jtyr) [inline filter_plugins/config_encoders.py:413]: Let's drop this part as it's very unlikely that somebody will ever need to set such value ;o)

**Review by jtyr — CHANGES_REQUESTED** [review]
Please could you cover this new feature with a new [test case](https://github.com/jtyr/ansible-config_encoder_filters/blob/master/tests/test_config_encoders.py#L97)?

File: filter_plugins/config_encoders.py (hunk 1 above)
Comment (aw1cks) [reply filter_plugins/config_encoders.py:413]: Removed now

**Comment by aw1cks** [comment]
Test cases added as of latest commit, everything is passing
Escape removed as well

Let me know what you think :)

**Comment by jtyr** [comment]
I think having only one use case covering "general" and "in section" would be sufficient.

**Review by jtyr — APPROVED** [review]
LGTM

**Pull request merged**